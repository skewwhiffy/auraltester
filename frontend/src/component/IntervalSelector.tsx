import React from 'react'

type OnChangeHandler = (
  clef: string,
  bottomNote: string,
  intervalQuality: string,
  intervalSize: string,
  keySignature: string
) => void

interface Props {
  onChange: OnChangeHandler
}

interface State {
  clef: string,
  bottomNote: string,
  intervalQuality: string,
  intervalSize: string,
  keySignature: string
}

class IntervalSelector extends React.Component<Props, State> {
  render() {
    return (
      <h1>Hello</h1>
    )
  }
}

export default IntervalSelector