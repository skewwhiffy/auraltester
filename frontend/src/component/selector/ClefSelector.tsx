import H1 from "../H1";
import Centre from "../Centre";
import FancyRadioButtons from "../FancyRadioButtons";

export default ClefSelector

interface Props {
  title?: string | null
  value?: string | null

  onChange(value: string): void
}

function ClefSelector({title, value, onChange}: Props) {
  return (
    <>
      {title && <Centre><H1>{title}</H1></Centre>}
      <Centre>
        <FancyRadioButtons value={value} onChange={onChange}>
          <FancyRadioButtons.Item>Bass</FancyRadioButtons.Item>
        </FancyRadioButtons>
      </Centre>
    </>
  )
}
