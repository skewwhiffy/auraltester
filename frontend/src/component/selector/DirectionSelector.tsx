import Centre from "../Centre"
import FancyRadioButtons from "../FancyRadioButtons"
import H1 from "../H1"
import {Direction} from "../../common/types";

interface DirectionSelectorProps {
    title?: string | null,
    value?: Direction | null,

    onChange(direction: Direction): void
}

const DirectionSelector = ({title, value, onChange}: DirectionSelectorProps) => {
    return (
        <>
            {title && <Centre><H1>{title}</H1></Centre>}
            <Centre>
                <FancyRadioButtons value={value} onChange={onChange}>
                    <FancyRadioButtons.Item value="ascending">Ascending</FancyRadioButtons.Item>
                    <FancyRadioButtons.Item value="descending">Descending</FancyRadioButtons.Item>
                </FancyRadioButtons>
            </Centre>
        </>
    )
}

export default DirectionSelector
