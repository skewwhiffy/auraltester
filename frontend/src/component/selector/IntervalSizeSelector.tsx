import Centre from "../Centre";
import FancyRadioButtons from "../FancyRadioButtons";
import {IntervalSize} from "../../common/types";
import H1 from "../H1";

interface Props {
    title?: string
    value?: IntervalSize

    onChange(value: IntervalSize): void
}

const IntervalSizeSelector = ({title, value, onChange}: Props) => {
    return (
        <>
            {title && <Centre><H1>{title}</H1></Centre>}
            <Centre>
                <FancyRadioButtons value={value} onChange={onChange}>
                    <FancyRadioButtons.Item value="1">Unison</FancyRadioButtons.Item>
                    <FancyRadioButtons.Item value="2">Second</FancyRadioButtons.Item>
                    <FancyRadioButtons.Item value="3">Third</FancyRadioButtons.Item>
                    <FancyRadioButtons.Item value="4">Fourth</FancyRadioButtons.Item>
                    <FancyRadioButtons.Item value="5">Fifth</FancyRadioButtons.Item>
                    <FancyRadioButtons.Item value="6">Sixth</FancyRadioButtons.Item>
                    <FancyRadioButtons.Item value="7">Seventh</FancyRadioButtons.Item>
                    <FancyRadioButtons.Item value="8">Octave</FancyRadioButtons.Item>
                </FancyRadioButtons>
            </Centre>
        </>
    )
}

export default IntervalSizeSelector