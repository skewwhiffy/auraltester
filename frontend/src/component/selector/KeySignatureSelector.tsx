import React from "react"
import { Col, Container, Row } from 'react-bootstrap'
import RadioButtons, { RadioButtonDefinition } from "../../util/RadioButtons"

type OnChangeHandler = (note: string) => void

interface Props {
  defaultValue?: string
  onChange: OnChangeHandler
}

interface State {
  keySignature: string
}

class KeySignatureSelector extends React.Component<Props, State>{
  constructor(props: Props) {
    super(props)
    this.state = {
      keySignature: props.defaultValue || 'C'
    }
    props.onChange(this.state.keySignature)
  }

  private keySignatureValues: Array<RadioButtonDefinition> = [
    { value: 'C#', label: 'C# / A# minor' },
    { value: 'F#', label: 'F# / D# minor' },
    { value: 'B', label: 'B / G# minor' },
    { value: 'E', label: 'E / C# minor' },
    { value: 'A', label: 'A / F# minor' },
    { value: 'D', label: 'D / B minor' },
    { value: 'G', label: 'G / E minor' },
    { value: 'C', label: 'None / C / A minor' },
    { value: 'F', label: 'F / D minor' },
    { value: 'Bb', label: 'Bb / G minor' },
    { value: 'Eb', label: 'Eb / C minor' },
    { value: 'Ab', label: 'Ab / F minor' },
    { value: 'Db', label: 'Db / Bb minor' },
    { value: 'Gb', label: 'Gb / Eb minor' },
    { value: 'Cb', label: 'Cb / Ab minor' },
  ]

  render() {
    return (
      <Container>
        <Row><Col>Key signature</Col></Row>
        <Row>
          <Col>
            <RadioButtons
              values={this.keySignatureValues}
              name='keySignature'
              defaultValue={this.state.keySignature}
              onChange={keySignature => this.props.onChange(keySignature)}
            />
          </Col>
        </Row>
      </Container>
    )
  }
}

export default KeySignatureSelector