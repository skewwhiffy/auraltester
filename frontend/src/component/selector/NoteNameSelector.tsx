import Centre from "../Centre";
import H1 from "../H1";
import FancyRadioButtons from "../FancyRadioButtons";
import {NoteName} from "../../common/types";

interface Props {
    title?: string,
    value?: NoteName,

    onChange(value: NoteName): void
}

const NoteNameSelector = ({title, value, onChange}: Props) => {
    return (
        <>
            {title && <Centre><H1>{title}</H1></Centre>}
            <Centre>
                <FancyRadioButtons
                    value={value}
                    onChange={onChange}
                >
                    <FancyRadioButtons.Item>A</FancyRadioButtons.Item>
                    <FancyRadioButtons.Item>B</FancyRadioButtons.Item>
                    <FancyRadioButtons.Item>C</FancyRadioButtons.Item>
                    <FancyRadioButtons.Item>D</FancyRadioButtons.Item>
                    <FancyRadioButtons.Item>E</FancyRadioButtons.Item>
                    <FancyRadioButtons.Item>F</FancyRadioButtons.Item>
                    <FancyRadioButtons.Item>G</FancyRadioButtons.Item>
                </FancyRadioButtons>
            </Centre>
        </>
    )
}

export default NoteNameSelector