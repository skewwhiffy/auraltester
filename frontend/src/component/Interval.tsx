import { useState } from 'react'
import { Container, Row } from 'react-bootstrap'
import { Notation } from 'react-abc'
import IntervalSelector from './selector/IntervalSelector'
import { api } from '../util'

interface State {
  abc: string
}

const Interval = (): JSX.Element => {
  const [state, setState] = useState<State>({ abc: '' })

  const intervalSelected = (
    clef: string,
    bottomNote: string,
    intervalQuality: string,
    intervalSize: string,
    keySignature: string
  ): void => {
    if ([clef, bottomNote, intervalQuality, intervalSize, keySignature].includes('')) {
      setState({
        ...state,
        abc: ''
      })
      return
    }

    (async () => {
      const response = await api.getInterval({
        clef,
        bottomNote,
        intervalQuality,
        intervalSize,
        keySignature
      })
      setState({
        ...state,
        abc: response.abc
      })
    })()
  }
  return (
    <Container>
      <Row>
        <Notation notation={state.abc} />
      </Row>
      <Row>
        <IntervalSelector onChange={intervalSelected} />
      </Row>
    </Container>
  )
}

export default Interval
