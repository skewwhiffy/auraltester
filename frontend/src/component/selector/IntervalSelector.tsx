import React from 'react'
import { Col, Container, Form, Row } from 'react-bootstrap'
import RadioButtons, { RadioButtonDefinition } from '../../util/RadioButtons'
import ClefSelector from './ClefSelector'
import NoteSelector from './NoteSelector'

type OnChangeHandler = (
  clef: string,
  bottomNote: string,
  intervalQuality: string,
  intervalSize: string,
  keySignature: string
) => void

export interface Props {
  onChange: OnChangeHandler
}

interface State {
  clef: string,
  bottomNote: string,
  intervalQuality: string,
  intervalSize: string,
  keySignature: string
}

class IntervalSelector extends React.Component<Props, State> {
  private keySignatureValues: Array<RadioButtonDefinition> = [
    { value: 'C#', label: 'C# / A# minor' },
    { value: 'F#', label: 'F# / D# minor' },
    { value: 'B', label: 'B / G# minor' },
    { value: 'E', label: 'E / C# minor' },
    { value: 'A', label: 'A / F# minor' },
    { value: 'D', label: 'D / B minor' },
    { value: 'G', label: 'G / E minor' },
    { value: 'C', label: 'None / C / A minor' },
    { value: 'F', label: 'F / D minor' },
    { value: 'Bb', label: 'Bb / G minor' },
    { value: 'Eb', label: 'Eb / C minor' },
    { value: 'Ab', label: 'Ab / F minor' },
    { value: 'Db', label: 'Db / Bb minor' },
    { value: 'Gb', label: 'Gb / Eb minor' },
    { value: 'Cb', label: 'Cb / Ab minor' },
  ]

  constructor(props: Props) {
    super(props);
    this.state = {
      clef: 'treble',
      bottomNote: 'C',
      intervalQuality: '',
      intervalSize: '',
      keySignature: 'C'
    }
  }

  render() {
    return (
      <Container>
        <Form>
          <Row>
            <Col>
              <ClefSelector
                defaultValue={this.state.clef}
                onChange={clef => this.onFormChange({ ...this.state, clef })}
              />
            </Col>
            <Col>
              <NoteSelector
                defaultValue={this.state.bottomNote}
                includeDoubleAccidentals={true}
                onChange={bottomNote => this.onFormChange({ ...this.state, bottomNote })}
              />
            </Col>
            <Col>
              <Container>
                <Row><Col>Interval</Col></Row>
                <Row>
                  <Col>
                    <RadioButtons
                      values={['diminished', 'minor', 'major', 'perfect', 'augmented'].map(value => ({ value }))}
                      name='intervalQuality'
                      defaultValue={this.state.intervalQuality}
                      onChange={intervalQuality => this.onFormChange({ ...this.state, intervalQuality })}
                    />
                  </Col>
                  <Col>
                    <RadioButtons
                      values={Array.from(Array(8).keys()).map(it => it + 1).map(it => ({ value: `${it}` }))}
                      name='intervalSize'
                      defaultValue={this.state.intervalQuality}
                      onChange={intervalSize => this.onFormChange({ ...this.state, intervalSize })}
                    />
                  </Col>
                </Row>
              </Container>
            </Col>
            <Col>
              <Container>
                <Row><Col>Key signature</Col></Row>
                <Row>
                  <Col>
                    <RadioButtons
                      values={this.keySignatureValues}
                      name='keySignature'
                      defaultValue={this.state.keySignature}
                      onChange={keySignature => this.onFormChange({...this.state, keySignature})}
                    />
                  </Col>
                </Row>
              </Container>
            </Col>
          </Row>
        </Form>
      </Container>
    )
  }

  private onFormChange = (newState: State) => {
    this.setState(newState)
    this.onStateChange(newState)
  }

  private onStateChange(state: State) {
    this.props.onChange(
      state.clef,
      state.bottomNote,
      state.intervalQuality,
      state.intervalSize,
      state.keySignature
    )
  }
}

export default IntervalSelector