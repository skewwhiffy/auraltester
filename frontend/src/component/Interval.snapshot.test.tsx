import Interval from './Interval'
import renderer from 'react-test-renderer'
import { Props as IntervalSelectorProps } from './selector/IntervalSelector'

const getResponse = {
  data: {
    abc: 'abc'
  }
}
jest.mock('react-abc', () => ({
  Notation: (props: any) => (<div>{JSON.stringify(props)}</div>)
}))
jest.mock('axios', () => ({
  get: async () => getResponse
}))
jest.mock('./selector/IntervalSelector', () => (props: IntervalSelectorProps) => (
  <div>INTERVAL SELECTOR {JSON.stringify(props)}</div>
))

describe('Interval snapshot', () => {
  it('matches snapshot', () => {
    const actual = renderer.create(
      <Interval />
    )

    expect(actual).toMatchSnapshot()
  })
})