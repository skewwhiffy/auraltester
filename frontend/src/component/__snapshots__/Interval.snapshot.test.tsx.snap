// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interval snapshot matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div>
      {"notation":""}
    </div>
  </div>
  <div
    className="row"
  >
    <div>
      INTERVAL SELECTOR 
      {}
    </div>
  </div>
</div>
`;
